corehq/apps/users/forms.py

```diff
@@ -572,6 +584,28 @@ def __init__(self, project, request_user, *args, **kwargs):
                 data_bind='value: location_id',
             )
 
+        self.two_stage_provisioning_enabled = TWO_STAGE_USER_PROVISIONING.enabled(self.domain)
+        if self.two_stage_provisioning_enabled:
+            confirm_account_field = crispy.Field(
+                'force_account_confirmation',
+                # data_bind='value: force_account_confirmation',
+            )
+            email_field = crispy.Field(
+                'email',
+                # data_bind='value: email',
+            )
+        else:
```

Comment:
It'd be more concise to remove these and update `_build_commcare_user` to default them to false/blank.

Reply:
Can you elaborate on this? This is used to hide the fields in the UI if the flag is not enabled. Just not sure how I could accomplish that inside `_build_commcare_user`?

Reply:
I meant that I don't see why these need to be included as hidden fields. To look at just email, couldn't you not include the `email` field if the flag is off, and then change [this line](https://github.com/dimagi/commcare-hq/blob/524562519cff51faa4845cb84d82a6ae159d784d/corehq/apps/users/views/mobile/users.py#L757) to `email = self.new_mobile_worker_form.cleaned_data.get('email, '')`?

Reply:
I think the (or at least a) issue with that is how the [entire form is dynamically generated](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/forms.py#L625-L709).

I guess you could refactor out the arguments to `Fieldset` into a list and then include/exclude them based on the flag and pass them in as `*args`. I can give that a shot and see how it feels if you think it's better.

Reply:
Got it, I had forgotten about the layout aspect, that's more annoying. Feel free to ignore this feedback.

> Side note: I really don't understand the value of crispy forms for something this JS heavy. It seems like this would be much simpler if we just used the form for the field/validaiton logic and rendered the whole thing in HTML / JS.

I agree with this, by the way. Really, all crispy forms with js bug me; seeing `data_bind` in python always strikes me as perverse.

Reply:
I'll still give it a shot - think it will work out better.

> I agree with this, by the way. Really, all crispy forms with js bug me; seeing data_bind in python always strikes me as perverse.

Cool, good to know. Is that a quorum to change it if I get inspired or do we need to run it by a wider group? :upside_down_face:

Reply:
If you get inspired, go for it. I don't think it's controversial to do a complex form - or really, any individual form - in HTML/JS. Getting rid of crispy altogether might be controversial. @biyeun is the only other person likely to have an opinion on this.